File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/editor/general/auto_import/AutoImportSettings.java
package com.github.alexandrecarlton.idea.settings.layout.editor.general.auto_import;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import org.immutables.value.Value;

import java.util.Optional;

@Value.Immutable
@JsonDeserialize(as = ImmutableAutoImportSettings.class)
public interface AutoImportSettings {
  Optional<JavaAutoImportSettings> java();
}

File: layout/src/main/kotlin/com/github/alexandrecarlton/idea/settings/dialog/editor/general/auto_import/AutoImportSettings.kt
package com.github.alexandrecarlton.idea.settings.dialog.editor.general.auto_import

import com.fasterxml.jackson.annotation.JsonProperty

data class AutoImportSettings(

    @JsonProperty("Java")
    val java: JavaAutoImportSettings? = null,

    @JsonProperty("Kotlin")
    val kotlin: KotlinAutoImportSettings? = null
)
